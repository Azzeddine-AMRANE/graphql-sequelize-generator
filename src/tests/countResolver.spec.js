const request = require('supertest')
const { deleteTables } = require('./testDatabase.js')
const { createServer, closeServer, resetDb } = require('./setupServer')

/**
 * Starting the tests
 */
describe('Test the count resolvers', () => {
  let server = null
  let trace = []
  let globalPreCallback = type => {
    trace.push(type)
  }

  beforeAll(async () => {
    server = await createServer({}, globalPreCallback)
  })

  afterAll(() => closeServer(server))

  beforeEach(async () => {
    trace = []
    await resetDb()
  })

  afterEach(async () => {
    await deleteTables()
  })

  it('Check that you can query a custom list resolver', async () => {
    const response = await request(server)
      .get(
        `/graphql?query=
          query getUserAndCount {
              user {
                id
              }
              userCount
          }`
      )
      .set('userId', 1)
    expect(response.body.data.user).not.toBeUndefined()
    expect(response.body.data).toMatchSnapshot('Users ids and count')
    expect(response.body.data.user.length).toBe(response.body.data.userCount)
    expect(trace).toMatchSnapshot()
  })

  it('Check that you can inject types attributes and associations into a random Type', async () => {
    const response = await request(server)
      .get(
        `/graphql?query=
          query getOddUsers {
            users: oddUser(limit:10) {
              id
              name
              company {
                id
              }
            }
          }`
      )
      .set('userId', 1)
    expect(response.body.errors).toBeUndefined()
    expect(response.body.data.users).not.toBeUndefined()
    expect(response.body.data).toMatchSnapshot('Users ids should only be odd.')
    expect(trace).toMatchSnapshot()
  })
})
