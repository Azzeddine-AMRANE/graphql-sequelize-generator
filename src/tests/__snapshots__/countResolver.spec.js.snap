// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the count resolvers Check that you can inject types attributes and associations into a random Type 1`] = `Array []`;

exports[`Test the count resolvers Check that you can inject types attributes and associations into a random Type: Users ids should only be odd. 1`] = `
Object {
  "users": Array [
    Object {
      "company": Object {
        "id": 1,
      },
      "id": 1,
      "name": "Test 1 c 2",
    },
    Object {
      "company": Object {
        "id": 1,
      },
      "id": 3,
      "name": "Test 3 c 2",
    },
    Object {
      "company": Object {
        "id": 1,
      },
      "id": 5,
      "name": "Test 5 c 2",
    },
    Object {
      "company": Object {
        "id": 1,
      },
      "id": 7,
      "name": "Test 7 c 2",
    },
    Object {
      "company": Object {
        "id": 1,
      },
      "id": 9,
      "name": "Test 9 c 2",
    },
    Object {
      "company": null,
      "id": 21,
      "name": "Test 21 c 2",
    },
    Object {
      "company": null,
      "id": 23,
      "name": "Test 23 c 2",
    },
    Object {
      "company": null,
      "id": 25,
      "name": "Test 25 c 2",
    },
    Object {
      "company": null,
      "id": 27,
      "name": "Test 27 c 2",
    },
    Object {
      "company": null,
      "id": 29,
      "name": "Test 29 c 2",
    },
  ],
}
`;

exports[`Test the count resolvers Check that you can query a custom list resolver 1`] = `
Array [
  "listGlobalBefore",
  "listGlobalBefore",
  "listBefore",
  "countBefore",
  "listAfter",
]
`;

exports[`Test the count resolvers Check that you can query a custom list resolver: Users ids and count 1`] = `
Object {
  "user": Array [
    Object {
      "id": 5,
    },
    Object {
      "id": 10,
    },
  ],
  "userCount": 2,
}
`;

exports[`Test the count resolvers Check that you cannot query the count resolver if it is not defined 1`] = `[Error: cannot GET /graphql?query=%0A%20%20%20%20%20%20%20%20%20%20query%20getLogCount%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20logCount%0A%20%20%20%20%20%20%20%20%20%20%7D (400)]`;

exports[`Test the count resolvers Check that you cannot query the count resolver if it is not defined 2`] = `Array []`;
